### ISSUE TITLE
Accessibility problems with the new room tiles

### ISSUE THREAD

**Issue by MarcoZehe** [vector-im/riot-web#11113]
### Description

The new implementation from matrix-org/matrix-react-sdk/pull/3499 introduces a variety of problems. While it improves things when navigating with the keyboard in application/forms mode with a screen reader, trying to read the room list in browse or reading mode (or virtual mode) poses problems. The list does not register well, and content children aren't picked up.

There are several reasons for this.

1. For one, the role "listbox" only accepts "option" children, see [the ARIA 1.1 spec](https://www.w3.org/TR/wai-aria-1.1/#listbox), but this list contains rich elements such as groups, and even buttons.
2. Option children can contan graphics, also as per the ARIA spec linked above, but not buttons or other rich content.

### Steps to reproduce

- With NVDA, load Riot.
- Make sure you are in at least two rooms.
- DownArrow through the page, and when you get to the rooms list, listen to what NVDA will say.
  - Expected: NVDA should say something like Rooms list, and then read the first room or the Persons group name if there are any DMs included.
  - Actual: NVDA says Rooms Listbox, followed by the Start Chat button.
- Press DownArrow again.
  - Expected: The next room or group should be spoken.
  - Actual: You are out of the list, and on the main container.

### Proposed solution

For all intents and purposes, this widget has many characteristics of a tree view, like found in File Explorer, Finder's folder list, etc.

To most cleanly solve this, I suggest to implement the [simple Tree View pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView), which would mean the following changes:

1. The room list gets a role of "tree", not "listbox".
2. The room items get a role of "treeitem".
3. The group roles remain, but the Person and Room buttons are turned into tree items as well, since they are what collapses and expands the sub tree items.
   * Consequentially, One must be able to arrow to these as well, they should not be skipped when arrowing up and down. Right now, these buttons aren't keyboard accessible.
   * In addition to Space or enter , right arrow should trigger the click action when collapsed, LeftArrow when expanded.
4. Make sure all other cases described in the design pattern are covered to match user expectations.

Logs being sent: yes/no

### Version information

- **Platform**: web

For the web app:

- **Browser**: Firefox 70 beta
- **OS**: Windows
- **URL**: riot.im/developriot-web?

**Comment by t3chguy** [comment]
> For one, the role "listbox" only accepts "option" children, see the ARIA 1.1 spec, but this list contains rich elements such as groups, and even buttons.

https://github.com/w3c/aria/pull/782 seems to allude that groups should be supported to group the option children.

**Comment by t3chguy** [comment]
So I had a play with Navigation Tree View and whilst it seems like it works, it'd mean removing Tab indexing from each of the Room Tiles (existing purposeful behavior) and also it seems a little overkill for something which is just groups of options, with exactly depth = 2

**Comment by MarcoZehe** [comment]
How do you propose the problems I outlined above will be solved then? Primarily the inability to expand and collapse the groups of buttons with the keyboard, and also do things like starting a new chat etc. with keyboard alone?

**Comment by t3chguy** [comment]
So I guess upon further investigation there are no better solutions, so will make some changes and get this into a custom build so that you can try it as I don't know enough about screen reader use to be able to fully vet it
